lib/__tests__/_createInsightsClient.test.ts

```diff
@@ -0,0 +1,32 @@
+import { createInsightsClient } from "../_createInsightsClient";
+
+describe("createInsightsClient", () => {
+  beforeEach(() => {
+    window.AlgoliaAnalyticsObject = undefined;
+  });
+
+  it("should return a function", () => {
+    expect(typeof createInsightsClient(() => {})).toBe("function");
+  });
+
+  it("registers itself to window", () => {
+    const aa = createInsightsClient(() => {});
+
+    expect(typeof window.AlgoliaAnalyticsObject).toBe("string");
+    expect(window[window.AlgoliaAnalyticsObject!]).toBe(aa);
+  });
+
+  it("doesn't register itself to window if key is already taken", () => {
```

Comment:
```suggestion
  it("doesn't register itself to window if`AlgoliaAnalyticsObject` is already taken", () => {
```

Reply:
I'll apply those changes to the `master` branch only